### PR TITLE
Get rid of Thread.exclusive deprecation, Upgrade to Sidekiq 4

### PR DESCRIPTION
**Pull request by ChaoticBoredom**
Don't know if this will be merged, but here it is.

Removes deprecation warning for Thread.exclusive, and updates gem specs to Sidekiq 4.

Had 0 deprecation warning for Sidekiq 4, only deprecation was the Thread.exclusive one.

Also bumps Gem version to 0.5.0

### CODE REVIEW

File: lib/sidekiq/throttler/rate_limit.rb [hunk 1]
```diff
@@ -166,6 +168,8 @@ def executions
 
       private
 
+      SEMAPHORE = Mutex.new
```
Comment (gevans) [inline lib/sidekiq/throttler/rate_limit.rb]: Can you rename this constant to `LOCK` and move it to the top of our class? That should help clarify its use.

File: lib/sidekiq/throttler/version.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module Sidekiq
   class Throttler
-    VERSION = '0.4.1'
+    VERSION = '0.5.0'
```
Comment (gevans) [inline lib/sidekiq/throttler/version.rb]: Please remove this change from your commit. I'll look at what's changed since the last release then tag and push a new version after merging your changes.

File: sidekiq-throttler.gemspec [hunk 3]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |gem|
   gem.require_paths = %w(lib)
 
   gem.add_dependency 'activesupport'
-  gem.add_dependency 'sidekiq', '>= 2.5', '< 4.0'
+  gem.add_dependency 'sidekiq', '>= 2.5', '< 5.0'
```
Comment (gevans) [inline sidekiq-throttler.gemspec]: Can you make this change in a separate commit?

**Comment by ChaoticBoredom** [comment]
@gevans by commit do you mean PR? If so I'll close this PR and create two separate ones.

**Comment by gevans** [comment]
Whichever is your preference. :smile: You could also `git rebase -i origin/master` to make a separate commit and I'll accept it.
